src/ray/core_worker/core_worker.cc

```diff
@@ -4236,26 +4248,33 @@ void CoreWorker::HandleDeleteSpilledObjects(rpc::DeleteSpilledObjectsRequest req
 void CoreWorker::HandleExit(rpc::ExitRequest request,
                             rpc::ExitReply *reply,
                             rpc::SendReplyCallback send_reply_callback) {
-  bool own_objects = reference_counter_->OwnObjects();
-  int64_t pins_in_flight = local_raylet_client_->GetPinsInFlight();
+  const bool own_objects = reference_counter_->OwnObjects();
+  const size_t num_pending_tasks = task_manager_->NumPendingTasks();
```

Comment:
Bug: we didn't consider pending tasks when we decide whether the core worker is idle or not however we consider it during drain and exit.

Reply:
Can we write a python test for this?